Samples/NetworkMediaStreamer/RTSPServer/src/RTSPAuthProvider.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+#include <pch.h>
+
+using namespace winrt::Windows::Security;
+using namespace winrt::Windows::Security::Credentials;
+using namespace Cryptography::Core;
+using namespace Cryptography;
+
+class CAuthProvider : public winrt::implements<CAuthProvider, IRTSPAuthProvider, IRTSPAuthProviderCredStore>
+{
+    std::string GetValueForParam(std::string message, std::string param)
+    {
+        auto idx = message.find(param);
+        std::string val;
+        if (idx != std::string::npos)
+        {
+            idx += param.size();
+            if (message[idx] == '\"') idx++;
+            auto endIdx = message.find_first_of("\" \r\n,", idx);
+            val = message.substr(idx, endIdx - idx);
+        }
+        return val;
+
+    }
+public:
+
+    STDMETHODIMP GetNewAuthSessionMessage(HSTRING* pAuthSessionMessage)

```

Comment:
>GetNewAuthSessionMessage [](start = 17, length = 24)

do any of these APIs need to be thread safe?

Reply:
this is already thread safe. They modify only function arguments. The members are read only by the API and only set/written during construction. 

---
In reply to: [491105603](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r491105603) [](ancestors = 491105603)